integration/presto/src/main/prestosql/org/apache/carbondata/presto/CarbonDataFileWriter.java

```diff
@@ -0,0 +1,188 @@
+  private final int fieldCount;
+  private final Object row;
+  private final SettableStructObjectInspector tableInspector;
+  private final List<StructField> structFields;
+  private final HiveWriteUtils.FieldSetter[] setters;
+
+  private boolean isCommitDone;
+
+  public CarbonDataFileWriter(Path outPutPath, List<String> inputColumnNames, Properties properties,
+      JobConf configuration, TypeManager typeManager) throws SerDeException {
+    requireNonNull(outPutPath, "path is null");
+    // take the outputPath same as location in compliance with the carbon store folder structure.
+    this.outPutPath = new Path(properties.getProperty("location"));
+    this.configuration = requireNonNull(configuration, "conf is null");
+    List<String> columnNames = Arrays
+        .asList(properties.getProperty(IOConstants.COLUMNS, "").split(CarbonCommonConstants.COMMA));
+    List<Type> fileColumnTypes =
+        HiveType.toHiveTypes(properties.getProperty(IOConstants.COLUMNS_TYPES, "")).stream()
+            .map(hiveType -> hiveType.getType(typeManager)).collect(toList());
+    this.fieldCount = columnNames.size();
+    this.serDe = new CarbonHiveSerDe();
+    serDe.initialize(configuration, properties);
+    this.tableInspector = (ArrayWritableObjectInspector) serDe.getObjectInspector();
+
+    this.structFields =
+        ImmutableList.copyOf(inputColumnNames.stream().map(tableInspector::getStructFieldRef)
+            .collect(toImmutableList()));
+
+    this.row = tableInspector.create();
+
+    this.setters = new HiveWriteUtils.FieldSetter[structFields.size()];
+    for (int i = 0; i < setters.length; i++) {
+      setters[i] = HiveWriteUtils.createFieldSetter(tableInspector, row, structFields.get(i),
+          fileColumnTypes.get(structFields.get(i).getFieldID()));
+    }
+    String encodedLoadModel = this.configuration.get(CarbonTableConfig.CARBON_PRESTO_LOAD_MODEL);
+    if (StringUtils.isNotEmpty(encodedLoadModel)) {
+      this.configuration.set(CarbonTableOutputFormat.LOAD_MODEL, encodedLoadModel);
+    }
+    try {
+      boolean compress = HiveConf.getBoolVar(this.configuration, COMPRESSRESULT);
+      Object writer =
+          Class.forName(MapredCarbonOutputFormat.class.getName()).getConstructor().newInstance();
+      this.recordWriter = ((MapredCarbonOutputFormat<?>) writer)
+          .getHiveRecordWriter(this.configuration, this.outPutPath, Text.class, compress,
+              properties, Reporter.NULL);
+    } catch (Exception e) {
+      LOG.error("error while initializing writer", e);
+      throw new RuntimeException("writer class not found");
+    }
+  }
+
+  @Override
+  public long getWrittenBytes() {
+    if (isCommitDone) {
+      try {
+        return outPutPath.getFileSystem(configuration).getFileStatus(outPutPath).getLen();
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
+      }
+    }
+    return 0;
+  }
+
+  @Override
+  public long getSystemMemoryUsage() {
+    return 0;
```

Comment:
actually, whoever extends the `HiveWriter`, needs to be implement these and cant throw exceptions because, `Hivewriterfactory `calls these methods for each writer registered, so if we throw exceptions here, insert job will fail. May be we need to implements these methods with stats support later once the insert is finished.

Reply:
added TODO and created Jira for tracking https://issues.apache.org/jira/browse/CARBONDATA-4038